### ISSUE TITLE
AttributeError: module 'jax.random' has no attribute 'KeyArray'

### ISSUE THREAD

**Issue by FlowerPurr** [scverse/scvi-tools#2530]
Hello, maintainer,

I am hoping you can assist me with a problem I am facing. I have installed the latest version of scvi-tools, but I am still encountering an error when running the example code. 

```python
import scanpy as sc
import scvi
from rich import print
from scib_metrics.benchmark import Benchmarker
```

```pytb
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In[4], line 1
----> 1 from scib_metrics.benchmark import Benchmarker

File ~/.local/lib/python3.9/site-packages/scib_metrics/__init__.py:4
      1 import logging
      2 from importlib.metadata import version
----> 4 from . import nearest_neighbors, utils
      5 from ._graph_connectivity import graph_connectivity
      6 from ._isolated_labels import isolated_labels

File ~/.local/lib/python3.9/site-packages/scib_metrics/nearest_neighbors/__init__.py:2
      1 from ._dataclass import NeighborsResults
----> 2 from ._jax import jax_approx_min_k
      3 from ._pynndescent import pynndescent
      5 __all__ = [
      6     "pynndescent",
      7     "jax_approx_min_k",
      8     "NeighborsResults",
      9 ]

File ~/.local/lib/python3.9/site-packages/scib_metrics/nearest_neighbors/_jax.py:7
      4 import jax.numpy as jnp
      5 import numpy as np
----> 7 from scib_metrics.utils import cdist, get_ndarray
      9 from ._dataclass import NeighborsResults
     12 @functools.partial(jax.jit, static_argnames=["k", "recall_target"])
     13 def _euclidean_ann(qy: jnp.ndarray, db: jnp.ndarray, k: int, recall_target: float = 0.95):

File ~/.local/lib/python3.9/site-packages/scib_metrics/utils/__init__.py:3
      1 from ._diffusion_nn import diffusion_nn
      2 from ._dist import cdist, pdist_squareform
----> 3 from ._kmeans import KMeans
      4 from ._lisi import compute_simpson_index
      5 from ._pca import pca

File ~/.local/lib/python3.9/site-packages/scib_metrics/utils/_kmeans.py:9
      6 import numpy as np
      7 from sklearn.utils import check_array
----> 9 from scib_metrics._types import IntOrKey
     11 from ._dist import cdist
     12 from ._utils import get_ndarray, validate_seed

File ~/.local/lib/python3.9/site-packages/scib_metrics/_types.py:9
      6 import scipy.sparse as sp
      8 NdArray = Union[np.ndarray, jnp.ndarray]
----> 9 IntOrKey = Union[int, jax.random.KeyArray]
     10 ArrayLike = Union[np.ndarray, sp.spmatrix, jnp.ndarray]

File ~/.local/lib/python3.9/site-packages/jax/_src/deprecations.py:53, in deprecation_getattr.<locals>.getattr(name)
     51   warnings.warn(message, DeprecationWarning, stacklevel=2)
     52   return fn
---> 53 raise AttributeError(f"module {module!r} has no attribute {name!r}")

AttributeError: module 'jax.random' has no attribute 'KeyArray'
```

#### Versions:

>
scvi-colab                0.12.0                
scvi-tools                1.1.1    
jax                           0.4.24
jaxlib                       0.4.24

**Comment by martinkim0** [comment]
Looks like this is coming from the scib-metrics package - could you file the issue in that repository please?

**Comment by FlowerPurr** [comment]
> Looks like this is coming from the scib-metrics package - could you file the issue in that repository please?

Thank you for your response. Following the error message and your guidance, I have identified that the issue lies with the _types.py file in the scib-metrics package. I have checked the GitHub repository for this file and found that the problem I encountered has already been addressed by you, the author.I am now facing the challenge of updating the _types.py file in the scib-metrics package to the correct version. Would you recommend downloading the new _types.py file and replacing the old one, or should I update the scib-metrics package itself? Could you please advise which method would be more appropriate?

**Comment by canergen** [comment]
Please update the scib-metrics package. Updating the package is the safer process and should be preferred over generating your own version of a package (combining files from different releases generates a new version of a software package without any guarantees).

**Comment by FlowerPurr** [comment]
> Please update the scib-metrics package. Updating the package is the safer process and should be preferred over generating your own version of a package (combining files from different releases generates a new version of a software package without any guarantees).

Thank you very much for your response. I will try to update the scib-metrics package to see if that resolves the issue.

**Comment by FlowerPurr** [comment]
> Please update the scib-metrics package. Updating the package is the safer process and should be preferred over generating your own version of a package (combining files from different releases generates a new version of a software package without any guarantees).

Hello, author. I have checked my scib-metrics package, and it is indeed the latest version 0.5.0. However, I am still encountering the same error. Could you please provide some suggestions on how to resolve this issue?

**Comment by FlowerPurr** [comment]
> Looks like this is coming from the scib-metrics package - could you file the issue in that repository please?

Thank you for your response. I have resolved the issue by referring to your handling of this problem on GitHub in the scib-metrics repository and modifying the source file. Additionally, I have a suggestion that it would be great to release a new version of scib-metrics to fix this bug.

**Comment by martinkim0** [comment]
Hi, so sorry about this. We haven't released since the patch - I'll go ahead and do this. For now, please install from the main branch of the package.

**Comment by martinkim0** [comment]
Hi, I've just released scib-metrics 0.5.1 with the fix. Thanks for your patience!
